lib/metadata/bucketfile/backend.js

```diff
@@ -184,121 +65,74 @@ class BucketFileInterface {
      * @param {function} cb - callback(err, db, attr)
      * @return {undefined}
      */
-    loadDBIfExistsNoRef(bucketName, log, cb) {
-        this.getBucketAttributesNoRef(bucketName, log, (err, attr) => {
-            if (err) {
-                return cb(err, null);
-            }
-            let db;
-            try {
-                db = this.client.sublevel(bucketName);
-                return cb(null, db, attr);
-            } catch (err) {
-                /* if the bucket is newly created the
-                   client cannot create sublevels without
-                   re-reading the manifest */
-                this.waitreco++;
-                this.reConnect(() => {
-                    this.waitreco--;
-                    try {
-                        db = this.client.sublevel(bucketName);
-                    } catch (err) {
-                        log.error('cannot make sublevel usable',
-                                  { error: err.stack });
-                        return cb(errors.InternalError, null);
-                    }
-                    return cb(null, db, attr);
-                });
-                return undefined;
-            }
-        });
-        return undefined;
-    }
-
     loadDBIfExists(bucketName, log, cb) {
-        this.ref();
-        this.loadDBIfExistsNoRef(bucketName, log, (err, db, attr) => {
+        this.getBucketAttributes(bucketName, log, (err, attr) => {
             if (err) {
-                this.unRef();
                 return cb(err);
             }
-            // we hold a ref here
-            return cb(err, db, attr);
+            const db = this.mdDB.openSub(bucketName);
```

Comment:
Even better, we can actually remove this code! (since it has no side-effects).

The other try/catch is not actually necessary since openSub does not throw, but I did it as a good practice to ensure we call the callback in case errors are thrown by some code below.